packages/nimble-components/src/table-column/menu-button/tests/table-column-menu-button.spec.ts

```diff
@@ -0,0 +1,538 @@
+                    name
+                );
+            });
+        });
+    });
+
+    describe('overflow', () => {
+        it('sets title on menu button when button text is ellipsized', async () => {
+            const longValue = 'long button text that is too long to fit in the space allocated';
+            table.style.width = '100px';
+            await table.setData([{ field: longValue }]);
+            await waitForUpdatesAsync();
+            const menuButton = pageObject.getMenuButton(0, 0)!;
+            menuButton.dispatchEvent(new MouseEvent('mouseover'));
+            await waitForUpdatesAsync();
+            expect(menuButton.getTitle()).toEqual(longValue);
+        });
+
+        it('does not set title when cell text is fully visible', async () => {
+            await table.setData([{ field: 'a' }]);
+            await waitForUpdatesAsync();
+            const menuButton = pageObject.getMenuButton(0, 0)!;
+            menuButton.dispatchEvent(new MouseEvent('mouseover'));
+            await waitForUpdatesAsync();
+            expect(menuButton.getTitle()).toEqual('');
+        });
+
+        it('removes title on mouseout of cell', async () => {
+            const longValue = 'long button text that is too long to fit in the space allocated';
+            table.style.width = '100px';
+            await table.setData([{ field: longValue }]);
+            await waitForUpdatesAsync();
+            const menuButton = pageObject.getMenuButton(0, 0)!;
+            menuButton.dispatchEvent(new MouseEvent('mouseover'));
+            await waitForUpdatesAsync();
+            menuButton.dispatchEvent(new MouseEvent('mouseout'));
+            await waitForUpdatesAsync();
+            expect(menuButton.getTitle()).toEqual('');
+        });
+    });
+
+    describe('table selection does not change', () => {
+        let menuButton: MenuButtonPageObject;
+
+        beforeEach(async () => {
+            table.idFieldName = 'id';
+            table.selectionMode = 'multiple';
+            await waitForUpdatesAsync();
+
+            await table.setData([{ id: '0', field: 'value' }]);
+            await waitForUpdatesAsync();
+            menuButton = pageObject.getMenuButton(0, 0)!;
+        });
+
+        it('when opening a menu button by clicking', async () => {
+            menuButton.clickMenuButton();
+            expect(menuButton.isOpen()).toBeTrue();
+            await waitForUpdatesAsync();
+            const selection = await table.getSelectedRecordIds();
+            expect(selection.length).toBe(0);
+        });
+
+        it('when opening a menu button by pressing Enter', async () => {
+            menuButton.pressEnterKey();
+            expect(menuButton.isOpen()).toBeTrue();
+            await waitForUpdatesAsync();
+            const selection = await table.getSelectedRecordIds();
+            expect(selection.length).toBe(0);
+        });
+    });
+
+    describe('menu button interactions', () => {
```

Comment:
My current thoughts are that 1 & 2 are overkill since that behavior is completely incapsulated within the menu/menu items. That behavior seems pretty far removed from the table column. But, I'll add a test for 3 to make sure that clicking on an open menu button fires all the expected events and doesn't change table selection.